### PR TITLE
Upgrade to Nuxt 3.12.4 and Vue 3.2.47

### PR DESCRIPTION
**Pull request by gcatanese**
Upgrade of the underlying frameworks to address a major vulnerability. 

This PR has involved a major refactoring as Nuxt3 is completely different:

- implemented in Typescript
- refactor all Vue pages
- implemented server-side APIs as API Routes
- removed unnecessary dependencies (Express, dotenv, crossenv)
- moved assets to `public` folder

### CODE REVIEW

File: server/api/webhooks/notifications.post.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    // Process the notification asynchronously based on the eventCode
+    await consumeEvent(notification); 
+
+    // Acknowledge event has been consumed by sending a 202 Accepted response
+    event.node.res.statusCode = 202;
+    return ''; // Return an empty body
+});
+
+// Process payload
+function consumeEvent(notification) {
```
Comment (Kwok-he-Chu) [inline server/api/webhooks/notifications.post.js:40]: Mark function as async, we use `await consumeEvent(notification)` earlier

File: server/api/webhooks/notifications.post.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    // Add item to DB, queue or different thread, we just log it for now
+    const merchantReference = notification.merchantReference;
+    const eventCode = notification.eventCode;
+    console.log('merchantReference:' + merchantReference + " eventCode:" + eventCode);
```
Comment (Kwok-he-Chu) [inline server/api/webhooks/notifications.post.js:44]: nit: Use consistent (double quotes) `"` during string concatenation

**Review by Kwok-he-Chu — CHANGES_REQUESTED** [review]
@gcatanese Nice work!

* [ ] `gitpod.yml` is still using port 8080, let's make sure it's changed to port 3000 
* [ ] Double check if the `npm run dev` still applies when spinning-up this repo in gitpod

File: plugins/adyen-web.client.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import '@adyen/adyen-web/dist/adyen.css'; // Importing CSS directly
+
+export default defineNuxtPlugin(() => {
```
[the remark this answers is not in the document]
Comment (gcatanese) [reply plugins/adyen-web.client.ts:3]: Not really. I have refactored to import the Adyen Drop-in CSS in the Nuxt configuration (no need to create a plugin).

**Comment by gcatanese** [comment]
Thanks @Kwok-he-Chu, I have addressed your comments, please have final look. Tnx

**Review by Kwok-he-Chu — APPROVED** [review]
Minor thing: When gitpod spins-up, the user is prompted with `Nuxt collects completely anonymous data about usage.` Users have to answer with `yes`/`no` in the terminal, can we somehow add this flag so that it defaults to `no`?

See screenshot:
image

**Comment by gcatanese** [comment]
> Minor thing: When gitpod spins-up, the user is prompted with `Nuxt collects completely anonymous data about usage.` Users have to answer with `yes`/`no` in the terminal, can we somehow add this flag so that it defaults to `no`?
> 
Yes, well spotted, I have disabled the `telemetry`

**Pull request merged**